### DISCUSSION TITLE
结合 Lagent 构建数据集训练 InternLM 模型，支持对 MMYOLO 等工具的调用 （15班）

### DISCUSSION THREAD

**Discussion by szhowardhuang** [InternLM/Tutorial#204] [project]
![image](https://github.com/InternLM/tutorial/assets/1407300/ff32dd1e-4f62-4652-ab91-960172e1b4fd)

### LLM Agent具备的3大优势：

语言理解和生成能力： LLM Agent在处理自然语言时表现出色。它们能够理解和生成复杂的语言结构，包括语法、语义和上下文相关性。

灵活性： LLM Agent可以用于多种不同的任务，包括文本生成、自然语言理解、对话系统等。这使得它们在不同领域和应用中具有广泛的灵活性。

自动化和效率： LLM Agent可以用于自动化许多文本处理任务，减轻人工劳动负担。它们能够快速生成文本、回答问题和执行其他语言相关的任务，提高工作效率。

我们有必要学习实践如何构建Agent。

### 我们大概要要做那些事情呢？

明确目标和任务： 确定Agent要解决的问题。我们目标放在图像识别。

选择技术栈： 预训练模型使用InternLm。

数据收集和准备： 收集并准备用于训练和测试的数据。待定

模型设计： 设计一个能够完成任务的模型架构。 待定

训练模型： 使用准备好的数据对预训练模型进行微调以适应我们的任务。待定

评估和调整： 对训练好的模型进行评估，检查其在测试数据上的性能。根据性能进行调整和优化。待定

集成到应用： 将训练好的模型集成到实际应用中。暂定OpenXLab。

![image](https://github.com/InternLM/tutorial/assets/1407300/ca67022e-355e-455d-bfe4-dea2dfb122ac)

### **这个项目涉及到数据收集清洗，模型微调，模型评估，应用实现及部署。 诚恳邀请有志者一起来动手。**

 同伴： 已满人

> [thkelper](https://github.com/thkelper)   

>  [xulinui](https://github.com/xulinui)

> [KJ-Huang-sz](https://github.com/KJ-Huang-sz)

> [SChaoWei](https://github.com/SChaoWei)

**Comment by Egber1t** [comment]
我这边有数据集（中医类），一起组队吗？微信：2057237078

**Reply by szhowardhuang** [reply]
建知识库， 第3节课正在实操，就不在大作业里面做了。

**Comment by gpt-dance** [comment]
同学， 我有意做这个项目，熟悉mmdet，seg，yolo系列工具，考虑组队吗？

**Reply by szhowardhuang** [reply]
ok

**Comment by xulinui** [comment]
同学，我也有意做这个项目，用mmrotate和mmseg分别做过两个项目，可以一起吗

**Reply by szhowardhuang** [reply]
ok

**Comment by gpt-dance** [comment]
> 同学，我也有意做这个项目，用mmrotate和mmseg分别做过两个项目，可以一起吗

可以留个联系方式？我们可以组队也可以讨论什么的。

**Reply by szhowardhuang** [reply]
微信号：huangzhengduo

**Reply by gpt-dance** [reply]
这个微信我加不上，我留下我的吧 purplish0301

**Reply by szhowardhuang** [reply]
ok

**Comment by KJ-Huang-sz** [comment]
请问下满人了吗，我也想做这个，之前做过mmdetection的项目

**Reply by gpt-dance** [reply]
加微信组队交流哈兄弟，楼主看见也可以加微信，purplish0301。

**Reply by KJ-Huang-sz** [reply]
ok，加你了

**Reply by szhowardhuang** [reply]
ok

**Reply by xulinui** [reply]
我也加了哈

**Reply by SChaoWei** [reply]
同学人满了吗

**Reply by szhowardhuang** [reply]
加你了

**Comment by SChaoWei** [comment]
您发给我的信件已收到。

**Reply by szhowardhuang** [reply]
加个微信吧 huangzhengduo

**Reply by SChaoWei** [reply]
这个微信加不上，你加我吧：scw_up

**Comment by aJupyter** [comment]
- 俺能学习一下嘛

**Reply by szhowardhuang** [reply]
请参考 https://github.com/szhowardhuang/VisionAgent ，进行中
